activesupport/lib/active_support/callbacks.rb

```diff
@@ -711,18 +717,18 @@ def skip_callback(name, *filter_list, &block)
 
           __update_callbacks(name) do |target, chain|
             filters.each do |filter|
-              callback = chain.find { |c| c.matches?(type, filter) }
+              while callback = chain.find { |c| c.matches?(type, filter) }
```

Comment:
This line is the only real change in this block, change from deleting the first to deleting all that match the type/filter combo.  We don't pass target here because skip_callbacks should skip all callbacks, not just the ones on the current class